### ISSUE TITLE
[Quantization]Why quantization is slower?

### ISSUE THREAD

**Issue by kibeenJung** [pytorch/glow#2147]
I tried to qunatize my model. But the inference time of quantized model is about 5 times slower than that of original model. 

I used softmax for multiclass classification. I refered to the following examples when writing quantization code. 

- fr2en.cpp : https://github.com/pytorch/glow/blob/master/examples/fr2en.cpp
- Loader.cpp : https://github.com/pytorch/glow/blob/master/tools/loader/Loader.cpp

I think I did profile and quantization process exactly. But problem occured. I don't know what the problem is. Can you help me?

** p.s : I used only one input file to make profile(profile.yaml). Can this be a problem?

**Comment by nadavrot** [comment]
We did not optimize the quantized cpu matmul and convolutions at all. Take a look at their implementation in "libjit_matmul.c"

**Comment by codezero759** [comment]
Thanks for your reply. How can I optimize the quantized cpu matmul (I never used convolution layers). In the example I mentioned earlier, there was no code for matmul.

**Comment by codezero759** [comment]
> We did not optimize the quantized cpu matmul and convolutions at all. Take a look at their implementation in "libjit_matmul.c"

Thanks for your reply. How can I optimize the quantized cpu matmul (I never used convolution layers). In the example I mentioned earlier, there was no code for matmul.

**Comment by nadavrot** [comment]
This was not a priority for us, but if you want to optimize quantized matrix multiplication you will need to look at the code in libjit_matmul.c and figure out how to optimize the quantized implementation for x86. You will need to map the operations into low-level x86 vector instructions, similar to how we've optimized the float32 implementation.
